packages/shared/src/components/modals/common/ModalHeader.tsx

```diff
@@ -85,7 +85,11 @@ export function ModalHeader({
         </ModalHeaderTitle>
       )}
       {shouldShowClose && (
-        <ModalClose className="hidden tablet:flex" onClick={onRequestClose} />
+        <ModalClose
+          type="button"
```

Comment:
This was submitting the form, which I am pretty sure is never intentional, so I added the type to make sure it never does.